SparkPost/2web2ui
# FE-1239, FE-1240 - Sending and Bounce empty states

### What Changed
 - Sending domain Empty State shows when no sending domains
 - Bounce domain Empty State shows when there are no sending and bounce domains
 - Sending domain Info Banner shows, but only until the user dismisses it
 - Bounce domain Info Banner shows, but only until the user dismisses it
 - Create domains page accepts param of: sending, bounce, and tracking
 - Segment tracks each tab through the new page prop, `props.empty?.trackingOnly`
 - Cypress test coverage added
 - ESLint rule to enforce page tracking calls are being done correctly - so the call to segment doesn't happen more than once on page load. 

### How To Test
 - Cypress tests in the listPage.spec.js should express the needed test scenarios, find the "empty state" tests and go through them to understand the states that the app can handle.
 - Load up the http://localhost:3100/domains in hibana with empty states turned off, then on.
 
 - Sending Domain Empty State shows when no sending domains **(with correct mock content)**
 - Bounce domain Empty State shows when there are no sending and bounce domains  **(with correct mock content)**
 - Sending Domain Info Banner shows, but only until the user dismisses it  **(with correct mock content)**
 - Bounce Domain Info Banner shows, but only until the user dismisses it  **(with correct mock content)**
 - Routing to the /domains/create state with the type param works and selects the type of domain to create
 - Cypress and unit tests pass and there is enough coverage
 - Page tracking calls segment, and only once for each tab/state

### To Do
- [x] Talk to UX about bounce empty state & banner (see comment)
- [x] Make UX updates provided by Tim
- [x] Add Cypress Test Coverage
